testing.md

```diff
@@ -0,0 +1,79 @@
+# Testing
+
+## Best Practices
+
+Writing good tests in javascript is similar to writing good tests in any other language. There are a few best practices that are more pertinent to javascript testing.
+
+### Mocking
+
+Mock any dependency with `sinon.js`. `sinon.js` is extremely easy to use.
+TODO
+
+### No AJAX calls
+TODO
```

Comment:
was going to give some examples on how to avoid ajax in tests, but agree i hope most people know that it's bad to do ajax in javascript testing